pkg/driver/node/credentialprovider/awsprofile/aws_profile.go

```diff
@@ -75,14 +76,8 @@ func CleanupAWSProfile(basepath string) error {
 	return nil
 }
 
-func writeAWSProfileFile(path string, content string) error {
-	err := os.WriteFile(path, []byte(content), awsProfileFilePerm)
-	if err != nil {
-		return err
-	}
-	// If the given file exists, `os.WriteFile` just truncates it without changing it's permissions,
-	// so we need to ensure it always has the correct permissions.
-	return os.Chmod(path, awsProfileFilePerm)
```

Comment:
[`renameio.WriteFile` does this](https://github.com/google/renameio/blob/v1.0.1/writefile.go#L30-L33) for us.